File: Day16/SqNoSum.py
class Solution(object):
    def judgeSquareSum(self, c):
        """
        :type c: int
        :rtype: bool
        """
        for a in range(0, int(c ** 0.5) + 1):
            b_sq = c - a * a
            b = int(b_sq ** 0.5)
            if b * b == b_sq:
                return True
        return False

if __name__ == "__main__":
    c = int(input())
    sol = Solution()
    print(sol.judgeSquareSum(c))

File: Day16/SqNoSum.java
import java.util.Scanner;

public class SqNoSum {
    public boolean judgeSquareSum(int c) {
        for (int a = 0; a <= Math.sqrt(c); a++) {
            int b_sq = c - (a * a);
            int b = (int) Math.sqrt(b_sq);
            if (b * b == b_sq) {
                return true;
            }
        }
        return false;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int c = scanner.nextInt();
        SqNoSum sol = new SqNoSum();
        System.out.println(sol.judgeSquareSum(c));
        scanner.close();
    }
}
